### PR TITLE
chore: refacto compile to handle rounding_threshold_bits

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/4351

### CODE REVIEW

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
could you also add the call to `check_rounding_threshold` in : 
- `build_quantized_module` (this method can be called separately from `_compile_torch_or_onnx_model`)
- `ONNXConverter`'s init (just to be safe)

**Comment by jfrery** [comment]
> could you also add the call to `check_rounding_threshold` in :
> 
>     * `build_quantized_module` (this method can be called separately from `_compile_torch_or_onnx_model`)
> 
>     * `ONNXConverter`'s init (just to be safe)

Yes good point about `build_quantized_module`. I have done the changes.

File: src/concrete/ml/common/utils.py [hunk 1]
```diff
@@ -605,3 +606,50 @@ def array_allclose_and_same_shape(
     assert isinstance(b, numpy.ndarray)
 
     return a.shape == b.shape and numpy.allclose(a, b, rtol, atol, equal_nan)
+
+
+def check_rounding_threshold(rounding_threshold_bits):
```
Comment (RomanBredehoft) [inline src/concrete/ml/common/utils.py:611]: did not realize that this was not only checking but also "building" the object. I would rename it to something else then, like `process_rounding_threshold_bits`

Comment (jfrery) [reply src/concrete/ml/common/utils.py:611]: Sure!

**Review by RomanBredehoft — APPROVED** [review]
great thanks a lot !

**Pull request merged**